### PR TITLE
Block Supports: Allow skipping spacing support serialization

### PR DESCRIPTION
**Pull request by aaronrobertshaw**
Related: https://github.com/WordPress/gutenberg/issues/24874

#### Background
As part of looking to add custom margins (gutters) to the column/s blocks, we run into an issue with the spacing block support styles being added as inline styles. This won't play nice with the CSS media queries that stack the columns at different viewports. These queries look to set a `flex-basis` for the column width taking into account margins so they stack correctly.

This highlighted the fact that the option to skip serialization of the padding and margin block support styles wasn't available. This PR allows for that skipping of serialization.

## How has this been tested?
Manually.

#### Test Notes
- The test instructions are just for testing that the serialization is skipped. This PR doesn't update any blocks.
- If testing with group block, block patterns with existing padding will throw block errors.

#### Test Instructions
1. Turn on `customMargin` under the `spacing` support in your theme.json
2. Update the Group block's `block.json` to enable both padding and margin support while skipping serialization - [gist](https://gist.github.com/aaronrobertshaw/5694d90c2aa1207b8abe1d30b2677ce9)
3. Create a new post and add a group block to it. (setting background color can help testing)
4. Adjust the block's padding and margin values in the sidebar
    The block's spacing shouldn't change although color selections etc should be applied and the block visually update
5. Switch to the code editor view
6. Confirm the padding and margin attributes reflect the values set
7. Confirm that the block's markup does not contain inline styles for either padding or margins
8. Save the post and view on the frontend. Confirm there are no padding or margin styles
9. Update the Site Title block's `block.json` file to skip serialization (or any other dynamic block with spacing support) - [gist](https://gist.github.com/aaronrobertshaw/316a54a2aeeaa88343de8701c7abc588)
10. Add Site Title block to the post and set padding and margin
11. Save the post and confirm that there are no padding and margin inline styles on the site title on the frontend

## Types of changes
New feature

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] I've tested my changes with keyboard and screen readers. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR (please manually search all `*.native.js` files for terms that need renaming or removal).

### CODE REVIEW

File: lib/block-supports/spacing.php
Comment (aaronrobertshaw) [inline lib/block-supports/spacing.php:39]: Removed `gutenberg_has_spacing_feature_support` as it was just a wrapper for `gutenberg_block_has_support`. Unless the support config is changed to allow all spacing support with a single boolean it isn't needed.

**Comment by apeatling** [comment]
@aaronrobertshaw Is this updating `gutenberg/packages/block-library/src/group/block.json`? I copied over the Gist there and rebuilt, but I'm still only seeing padding:

Screen Shot 2021-05-20 at 4 52 45 PM

**Comment by aaronrobertshaw** [comment]
> Is this updating `gutenberg/packages/block-library/src/group/block.json`? I copied over the Gist there and rebuilt, but I'm still only seeing padding:

My apologies @apeatling, I missed that I'd also updated `experimental-default-theme.json` to toggle on `customMargin` under the spacing support. I'll fix the instructions.

I've just force pushed a rebase resolving the previous conflict in `style.js` after the change for skipping typography support serialization was merged. You might also need to update your local branch.

**Review by stacimc — APPROVED** [review]
This looks great to me, and +1 for adding this support -- it will also be useful for adding [padding to the button block](https://github.com/WordPress/gutenberg/pull/31774/files).

In the testing instructions, in step 4:
> Block should still visually update

I think this is saying that the block should still visually update for _other_ controls like color, but padding/margin changes don't visually reflect since you've already enabled skip serialization in an earlier step. This tests correctly and is very clear from watching the testing gif you've supplied, just wanted to clarify for others testing!

**Comment by aaronrobertshaw** [comment]
> I think this is saying that the block should still visually update for other controls like color, but padding/margin changes don't visually reflect since you've already enabled skip serialization in an earlier step.

Thanks @stacimc, you are correct, I've updated the test instructions. Appreciate you taking the time to review this. 

> it will also be useful for adding padding to the button block.

As soon as we get the required approval I'll merge this.

**Review by glendaviesnz — APPROVED** [review]
Tested well for me, and code looks good in comparison to other blocks that already have this opt out.

**Pull request merged**
